qemu/tests/x86_cpu_model.py

```diff
@@ -6,6 +6,37 @@
 from virttest import error_context, utils_misc, env_process
 
 
+def check_flags(params, flags, test, session=None):
+    """
+    Check cpu flags on host or guest.(only for Linux now)
+    :param params: Dictionary with the test parameters
+    :param flags: checked flags
+    :param test: QEMU test object
+    :param session: guest session
+    """
+    if session:
+        if params["os_type"] == "linux":
+            error_context.context("Check cpu flags inside guest", logging.info)
+            cmd = params["check_flag_cmd"]
+            out = session.cmd_output(cmd).split()
+            missing = [f for f in flags.split() if f not in out]
+            if missing:
+                test.fail("Flag %s not in guest" % missing)
+            no_flags = params.get("no_flags")
+            if no_flags:
+                err_flags = [f for f in no_flags.split() if f in out]
+                if err_flags:
+                    test.fail("Flag %s should not be present in guest" % err_flags)
+    else:
```

Comment:
There are some duplicate code in the if and else block, I'd prefer to eliminate them.

Reply:
it seems not clearer than this way, a little change updated and please help to review, thanks.

Reply:
@nanliu-r What about this:
```
func = process.system_output
    if session:
        func = session.cmd_output
out = func(cmd).decode().split()
missing = ...
```

Reply:
thanks and update, please help to check.

Reply:
Test pass: (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.memory_protection_key.q35: PASS (71.52 s)

Reply:
I guess the test can't pass with windows guest.

Reply:
> I guess the test can't pass with windows guest.

update please help to check. thanks.